app/Child.php

```diff
@@ -23,6 +39,151 @@ class Child extends Model
     protected $hidden = [
     ];
 
+
+    protected $dates = [
+        'created_at',
+        'updated_at',
+        'dob'
+    ];
+
+    protected $appends = [
+        'entitlement',
+    ];
+
+    /**
+     * Calculates and returns the age in Years and Months
+     *
+     * @param string $format
+     * @return string
+     */
+    public function getAgeString($format = '%y yr, %m mo')
+    {
+        return $this->dob->diff(Carbon::now())->format($format);
+    }
+
+    /**
+     * Returns the DoB as a string
+     *
+     * @param string $format
+     * @return string
+     */
+    public function getDobAsString($format = 'M Y')
+    {
+        return $this->dob->format($format);
+    }
+
+    /**
+     * Calculates the School start date for a Child
+     * If a Child is born before september, 4 years ahead
+     * Else 5 years ahead
+     *
+     * @return Carbon
+     */
+    public function calcSchoolStart()
+    {
+        // september is month 9
+        if ($this->dob->month < 9) {
+            $years = 4;
+        } else {
+            $years = 5;
+        }
+        $school_year = $this->dob->addYears($years)->year;
+        return Carbon::createFromDate($school_year, 9)->startOfMonth();
+    }
+
+    /**
+     * Get an array that holds
+     * Notices - array of Notice constants
+     * Credits - array of Credit constants
+     * Eligibility - status of child on scheme
+     * Vouchers - total vouchers this child is permitted
+     *
+     * These can be used in voucher multipliers
+     *
+     * @return array
+     */
+
+    public function getStatus()
+    {
+
+        $today = Carbon::today();
+
+        $notices = [];
+        $credits = [];
+
+        $eligibility = "Ineligible";
+
+        if (!$this->born) {
+            // Regardless of age, if you are unborn, you count as a pregnancy and get not credits
+            // Even positive ages! This is a process thing
+
+            $eligibility = "Pregnancy";
+
+            // Calculate notices
+            $is_almost_born = ($today->diffInMonths($this->dob) < 1) && ($this->dob->isFuture());
+            $is_overdue = ($today->diffInMonths($this->dob) > 1) && ($this->dob->isPast());
```

Comment:
Should this be diff in days?

Reply:
The criteria is "in a month or so". It's a bit fluffy because they don't actually want to be too accurate about it for sensetivity reasons.